### PR TITLE
AP_Motors: Add current limiting to 6DOF motors for Sub

### PR DESCRIPTION
**Pull request by jaxxzer**
This affects Sub only.

### CODE REVIEW

File: libraries/AP_Motors/AP_Motors6DOF.cpp [hunk 1]
```diff
@@ -338,6 +343,42 @@ void AP_Motors6DOF::output_armed_stabilizing()
...
         }
     }
+
+	// Current limiting
+    if (_batt_current_max <= 0.0f) {
+        return;
+    }
+
+    static float _output_limited = 1.0f;
+    static float _batt_current_last = 0.0f;
```
Comment (OXINARF) [inline libraries/AP_Motors/AP_Motors6DOF.cpp]: Can you change these to be in the class instead? It's the preferred way now.

**Comment by jaxxzer** [comment]
Yes, I was just about to fix it up. I dug this out of a branch from a while back.

File: libraries/AP_Motors/AP_Motors6DOF.cpp (hunk 1 above)
Comment (jaxxzer) [reply libraries/AP_Motors/AP_Motors6DOF.cpp]: done

File: libraries/AP_Motors/AP_Motors6DOF.cpp [hunk 2]
```diff
@@ -338,6 +344,43 @@ void AP_Motors6DOF::output_armed_stabilizing()
             }
         }
     }
+
+	// Current limiting
+    if (_batt_current_max <= 0.0f) {
+        return;
+    }
+
+    AP_BattMonitor &battery = AP::battery();
```
Comment (amilcarlucas) [inline libraries/AP_Motors/AP_Motors6DOF.cpp]: shouldn't some of these be defined `const` just to be on the safe side ?

Comment (jaxxzer) [reply libraries/AP_Motors/AP_Motors6DOF.cpp]: Done. None of the others should be const.

**Comment by jaxxzer** [comment]
@amilcarlucas Done. None of the others are const.

**Comment by jaxxzer** [comment]
Are we good here? This is NFC for everything but sub.

**Comment by jaxxzer** [comment]
Bump.

**Comment by jaxxzer** [comment]
[ ] Add override keyword to virtual method
[ ] Add support for battery index
[ ] Add check for has current monitoring

**Pull request merged**